### PR TITLE
[cleanup] Make more playlist extractions lazy

### PR DESCRIPTION
**Pull request by seproDev**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Make playlist extraction in multiple extractors lazy by using generator expressions instead of list comprehension.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

**Comment by grqz** [comment]
<details><summary>there are more</summary>

```code-search
# Query: \\.playlist_result[\\s\\n]*\\([\\s\\n]*\\[
# Flags: RegExp
# ContextLines: 3

14 results - 14 files

yt_dlp/extractor/abc.py:
  138              urls_info = [urls_info]
  139  
  140          if youtube:
  141:             return self.playlist_result([
  142                  self.url_result(url_info['url']) for url_info in urls_info])
  143  
  144          formats = []

yt_dlp/extractor/banbye.py:
  26  
  27      def _extract_playlist(self, playlist_id):
  28          data = self._download_json(f'{self._API_BASE}/playlists/{playlist_id}', playlist_id)
  29:         return self.playlist_result([
  30              self.url_result(f'{self._VIDEO_BASE}/{video_id}', BanByeIE)
  31              for video_id in data['videoIds']], playlist_id, data.get('name'))
  32  

yt_dlp/extractor/bbc.py:
  1444          entries.extend(re.findall(
  1445              rf'setPlaylist\("({EMBED_URL})"\)', webpage))
  1446          if entries:
  1447:             return self.playlist_result(
  1448:                 [self.url_result(entry_, 'BBCCoUk') for entry_ in entries],
  1449                  playlist_id, playlist_title, playlist_description)
  1450  
  1451          # Multiple video article (e.g. http://www.bbc.com/news/world-europe-32668511)

yt_dlp/extractor/cineverse.py:
  135  
  136          season = traverse_obj(pageprops, ('seasonEpisodes', ..., 'episodes', lambda _, v: v['item_id'] and v['title']))
  137          if season:
  138:             return self.playlist_result([item_result(ep) for ep in season], playlist_id=series_id,
  139                                          playlist_title=traverse_obj(pageprops, ('itemDetailsData', 'title')))
  140          return item_result(pageprops['itemDetailsData'])

yt_dlp/extractor/duboku.py:
  239                  f'Cannot find {fragment}' if fragment else 'Cannot extract playlist')
  240  
  241          # return url results
  242:         return self.playlist_result([
  243              self.url_result(
  244                  urllib.parse.urljoin('https://w.duboku.io', x['href']),
  245                  ie=DubokuIE.ie_key(), video_title=x.get('title'))

yt_dlp/extractor/francetv.py:
  437  
  438          dailymotion_urls = tuple(DailymotionIE._extract_embed_urls(url, webpage))
  439          if dailymotion_urls:
  440:             return self.playlist_result([
  441                  self.url_result(dailymotion_url, DailymotionIE.ie_key())
  442                  for dailymotion_url in dailymotion_urls])
  443  

yt_dlp/extractor/gamejolt.py:
  410          game_id = self._match_id(url)
  411          game_overview = self._call_api(
  412              f'web/discover/games/overview/{game_id}', game_id, note='Downloading soundtrack info', errnote='Unable to download soundtrack info')
  413:         return self.playlist_result([{
  414              'id': str_or_none(song.get('id')),
  415              'title': str_or_none(song.get('title')),
  416              'url': str_or_none(song.get('url')),

yt_dlp/extractor/rtvcplay.py:
  170                  if not podcast_episodes:
  171                      raise ExtractorError('Could not find asset_id nor program playlist nor podcast episodes')
  172  
  173:                 return self.playlist_result([
  174                      self.url_result(episode['file'], url_transparent=True, **traverse_obj(episode, {
  175                          'title': 'title',
  176                          'description': ('description', {clean_html}),

yt_dlp/extractor/s4c.py:
   97                  'show_prog_in_series': 'Y',
   98              }, note='Downloading series details JSON')
   99  
  100:         return self.playlist_result(
  101:             [self.url_result(f'https://www.s4c.cymru/clic/programme/{episode_id}', S4CIE, episode_id)
  102               for episode_id in traverse_obj(series_details, ('other_progs_in_series', ..., 'id'))],
  103              series_id, traverse_obj(series_details, ('full_prog_details', 0, 'series_title', {str})))

yt_dlp/extractor/sbscokr.py:
  164                  'currentCount': '10000',
  165              })
  166  
  167:         return self.playlist_result(
  168:             [self.url_result(f'https://allvod.sbs.co.kr/allvod/vodEndPage.do?mdaId={video_id}', SBSCoKrIE)
  169               for video_id in traverse_obj(details, ('list', ..., 'mdaId'))], program_id)
  170  
  171  

yt_dlp/extractor/seznamzpravy.py:
  149          title = info.get('title') or self._og_search_title(webpage, fatal=False)
  150          description = info.get('description') or self._og_search_description(webpage)
  151  
  152:         return self.playlist_result([
  153              self.url_result(entry_url, ie=SeznamZpravyIE.ie_key())
  154              for entry_url in SeznamZpravyIE._extract_embed_urls(url, webpage)],
  155              article_id, title, description)

yt_dlp/extractor/testurl.py:
  37  
  38          testcases = tuple(extractor.get_testcases(True))
  39          if num == 'all':
  40:             return self.playlist_result(
  41:                 [self.url_result(tc['url'], extractor) for tc in testcases],
  42                  url, f'{extractor.IE_NAME} tests')
  43          try:
  44              tc = testcases[int(num or 0)]

yt_dlp/extractor/youtube.py:
  6873          return any(self._extract_tab_id_and_name(tab)[0] == tab_id for tab in tabs)
  6874  
  6875      def _empty_playlist(self, item_id, data):
  6876:         return self.playlist_result([], item_id, **self._extract_metadata_from_tabs(item_id, data))
  6877  
  6878      @YoutubeTabBaseInfoExtractor.passthrough_smuggled_data
  6879      def _real_extract(self, url, smuggled_data):

yt_dlp/extractor/zaiko.py:
  137          webpage = self._download_real_webpage(url, ticket_id)
  138          eticket = self._parse_vue_element_attr('eticket', webpage, ticket_id)
  139  
  140:         return self.playlist_result(
  141:             [self.url_result(stream, ZaikoIE) for stream in traverse_obj(eticket, ('streams', ..., 'url'))],
  142              ticket_id, **traverse_obj(eticket, ('ticket-details', {
  143                  'title': 'event_name',
  144                  'thumbnail': 'event_img_url',
```


</details>

**Comment by seproDev** [comment]
I didn't include those since no network requests are being made

File: yt_dlp/extractor/nytimes.py [hunk 1]
```diff
@@ -343,7 +343,7 @@ def _real_extract(self, url):
         if media_ids:
             media_ids.append(lead_video_id)
             return self.playlist_result(
-                [self._extract_video(media_id) for media_id in media_ids], page_id, title, description)
+                (self._extract_video(media_id) for media_id in media_ids), page_id, title, description)
```
Comment (Grub4K) [inline yt_dlp/extractor/nytimes.py:346]:
```suggestion
                map(self._extract_video, media_ids), page_id, title, description)
```

**Review by Grub4K — APPROVED** [review]

File: yt_dlp/extractor/brightcove.py [hunk 2]
```diff
@@ -935,8 +936,8 @@ def extract_policy_key():
 
         if content_type == 'playlist':
             return self.playlist_result(
-                [self._parse_brightcove_metadata(vid, vid.get('id'), headers)
-                 for vid in json_data.get('videos', []) if vid.get('id')],
+                (self._parse_brightcove_metadata(vid, vid.get('id'), headers)
```
Comment (bashonly) [inline yt_dlp/extractor/brightcove.py:939]:
```suggestion
                (self._parse_brightcove_metadata(vid, vid['id'], headers)
```

**Review by bashonly — APPROVED** [review]

**Pull request merged**